openpype/lib/applications.py

```diff
@@ -1038,28 +1038,30 @@ def get_app_environments_for_context(
         "name": asset_name
     })
 
-    # Prepare app object which can be obtained only from ApplciationManager
-    app_manager = ApplicationManager()
-    app = app_manager.applications[app_name]
-
     # Project's anatomy
     anatomy = Anatomy(project_name)
 
-    data = EnvironmentPrepData({
+    prep_dict = {
         "project_name": project_name,
         "asset_name": asset_name,
         "task_name": task_name,
 
-        "app": app,
-
         "dbcon": dbcon,
         "project_doc": project_doc,
         "asset_doc": asset_doc,
 
         "anatomy": anatomy,
 
         "env": env
-    })
+    }
+
+    if app_name:
+        # Prepare app object which can be obtained only from ApplicationManager
+        app_manager = ApplicationManager()
+        app = app_manager.applications[app_name]
+        prep_dict["app"] = app
```

Comment:
This must crash on `prepare_host_environments` if there is not `"app"` key in `data`. Is this even used?

Reply:
Mixed up commit, reverted.